src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -73,9 +73,23 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
 				Integer i = 0;
 				for (SObject so : newlist) {
 					Affiliation__c affl = (Affiliation__c)so;
+          
+					// Get the 'Primary Affl Field' value(could be Lookup Label or the Lookup API name) that is stored in 'Affl Mapping' Custom Settings
+					String lookupLabelOrAPIName = afflMapper.accRecTypeToContactLabel.get(affl.Affiliation_Type__c);
 
-					String lookupFieldLabel = afflMapper.mapAccRecTypeToContactLabel.get(affl.Affiliation_Type__c);
-					String lookupFieldName = afflMapper.contactLabelNames.get(lookupFieldLabel);
+					// Get the API name of the lookup field if lookupLabelOrAPIName is actually a Label and assig it to lookupFieldAPIName
+					String lookupFieldAPIName = afflMapper.contactLabelNames.get(lookupLabelOrAPIName);
+          
+					// If lookupFieldAPIName of the Contact is not found, check if lookupLabelOrAPIName is actually a field API Name, if yes assign it to lookupFieldAPIName.
+					// This way the API name can be used to populate primay Affiliation lookup on contact even when language is not English
+						
+                    if ((lookupLabelOrAPIName != NULL && lookupLabelOrAPIName != '') && (lookupFieldAPIName == '' || lookupFieldAPIName == NULL)) {
```

Comment:
@skosgi1 Just a quick formatting comment on this - could you run the "Convert Indentation to Spaces" command in VS Code on this file, like we were talking about in our Quip chat today? The indentation looks funny when looking at this in Github.

Reply:
@robertbwatson done!